### PR TITLE
General api

### PR DESCRIPTION
**Pull request by narekhovhannisyan**
## Motivation

Integrate General API endpoints into Mailtrap Node.js SDK.
Cover changes with unit tests.

## Changes

- Implemented Accounts, AccountAccesses, Permissions APIs.
- Covered resources below with unit tests.

### CODE REVIEW

File: src/lib/api/resources/Permissions.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  /**
+   * Manage user or token permissions. For this endpoint, you should send an array of objects (in JSON format) as the body of the request.
+   * - `resource_type` can be account, billing, project, inbox or mailsend_domain.
+   * - `resource_id` the ID of the resource
+   * - `access_level` can be admin or viewer or their numbers 100 and 10 respectively
+   * - `_destroy(optional)` a boolean. If true, instead of creating/updating the permission, it destroys it
+   *
+   * If you send a combination of resource_type and resource_id that already exists, the permission is updated. If the combination doesn’t exist, the permission is created.
+   */
+  public async manageUserOrToken(
```
Comment (VladimirTaytor) [inline src/lib/api/resources/Permissions.ts:35]: I'm not sure `manageUserOrToken` would be a descriptive enough naming, what about
- `upsertPermissions`
- `bulkUpdatePermissions`
- maybe there are other ideas?

File: src/lib/api/resources/AccountAccesses.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    this.accountId = accountId;
+    this.accountAccessesURL = `${GENERAL_ENDPOINT}/api/accounts/${this.accountId}/account_accesses`;
+  }
+
+  /**
+   * Get list of account accesses for which specifier_type is User or Invite.
+   * You have to have account admin/owner permissions for this endpoint to work.
+   * If you specify project_ids, inbox_ids or domain_uuids, the endpoint will return account accesses for these resources.
+   */
+  public async listUserAndInviteAccountAccesses(
```
Comment (VladimirTaytor) [inline src/lib/api/resources/AccountAccesses.ts:32]: Let's maybe not specify implementation details for this route
- `listAccountAccesses` should be enough, we might start returning not only Users and Invites :)

File: src/lib/api/resources/Permissions.ts (hunk 1 above)
Comment (narekhovhannisyan) [reply src/lib/api/resources/Permissions.ts:35]: @VladimirTaytor got the name from API documentation. But as needed can be updated.

**Review by mklocek — APPROVED** [review]

File: src/lib/api/resources/Accounts.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  constructor(client: AxiosInstance) {
+    this.client = client;
+    this.accountsURL = `${GENERAL_ENDPOINT}/api/accounts`;
+  }
+
+  /**
+   * Get a list of your Mailtrap accounts.
+   * @returns Returns the list of accounts to which the API token has access.
+   * access_levels can return 1000 (account owner), 100 (admin), 10 (viewer).
```
Comment (leonid-shevtsov) [inline src/lib/api/resources/Accounts.ts:23]: these should be constants or an enum?

Comment (narekhovhannisyan) [reply src/lib/api/resources/Accounts.ts:23]: will introduce them in return type instead

**Pull request merged**